### PR TITLE
User experience dashboard

### PR DESCRIPTION
**Pull request by nicklin37**
#WHAT: Terraform support to create example dashboard for User Experience, contains metrics that are relevant to user interaction with the application

#WHY: We want to provide dashboard that reveals metrics about the demo application as whole

#HOW: Additional terraform config file created alongside initial provisioning terraform

### CODE REVIEW

File: terraform/monitoring/00_provider.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# that were populated appropriately at runtime.
+
+provider "google" {
+  # pin provider to 3.23.0
+  version = ">=3.23.0"
+
+  # credentials = "/path/to/creds.json"
+  # project = "project-id"
+  # region = "default-region"
+  # zone = "default-zone"
```
Comment (simonz130) [inline terraform/monitoring/00_provider.tf:32]: Should we come up with defaults?

File: terraform/monitoring/02_dashboards.tf [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    "columns": "2",
+    "widgets": [
+      {
+        "title": "HTTP Request Count by Method",
+        "xyChart": {
+          "dataSets": [
+            {
+              "timeSeriesQuery": {
+                "timeSeriesFilter": {
+                  "filter": "metric.type=\"custom.googleapis.com/opencensus/opencensus.io/http/server/request_count_by_method\"",
```
Comment (simonz130) [inline terraform/monitoring/02_dashboards.tf:32]: FYi @gzhao408, if Nic's PR is merged before your work on OC->OT, we will need to update the name of the metric after we move to OT

Comment (simonz130) [reply terraform/monitoring/02_dashboards.tf:32]: There are a few more places like that in terraform monitoring

**Comment by simonz130** [comment]
@nicklin37 - nit remark about defaults but otherwise it's ready to go. Please rebase and merge.

File: terraform/monitoring/00_provider.tf (hunk 1 above)
Comment (nicklin37) [reply terraform/monitoring/00_provider.tf:32]: This file will eventually get merged with the one in /sandbox/terraform so I don't think it makes sense to create defaults here. I think the comments are here just to demonstrate the option.

Comment (nicklin37) [reply terraform/monitoring/00_provider.tf:32]: If anything, we might need to add defaults to the provisioning script provider but that will be for @jiang95-dev

**Comment by gzhao408** [comment]
It looks like you merged master into your branch instead of rebasing your branch on top of master. I recommend:
`git reset --hard 107bdda` to get HEAD back to your top commit.
`git rebase master` to rebase your branch on top of recent master. This puts Dan's commits before yours (you want your updates to be on top of master).
`git push -f` to update remote (here). You need to force push because the parent commit of your base commit is different and thus it's not a simple push.
In general, don't merge anything except into master.

File: terraform/monitoring/02_dashboards.tf (hunk 2 above)
Comment (gzhao408) [reply terraform/monitoring/02_dashboards.tf:32]: Got it, will remember to replace these lines when I change the metrics

**Comment by nicklin37** [comment]
Rebased and now merging.

**Pull request merged**